buster/validators/base.py

```diff
@@ -12,60 +11,79 @@
 
 
 class Validator:
-    def __init__(self, validator_cfg):
-        self.cfg = validator_cfg
-
-        self.embedding_model = self.cfg["embedding_model"]
-        self.unknown_threshold = self.cfg["unknown_threshold"]
-        self.unknown_prompt = self.cfg["unknown_prompt"]
-
-        # set the unk. embedding
-        self.unk_embedding = self.get_embedding(self.unknown_prompt, engine=self.embedding_model)
-
-    @property
-    def unk_embedding(self):
-        return self._unk_embedding
-
-    @unk_embedding.setter
-    def unk_embedding(self, embedding):
-        logger.info("Setting new UNK embedding...")
-        self._unk_embedding = embedding
-        return self._unk_embedding
+    def __init__(self, embedding_model: str, unknown_threshold: float, unknown_prompt: str, use_reranking: bool):
+        self.embedding_model = embedding_model
+        self.unknown_threshold = unknown_threshold
+        self.unknown_prompt = unknown_prompt
+        self.use_reranking = use_reranking
 
+    @staticmethod
     @lru_cache
-    def get_embedding(self, query: str, engine: str):
+    def get_embedding(query: str, engine: str):
         logger.info("generating embedding")
         return get_embedding(query, engine=engine)
 
-    def check_sources_used(self, completion: Completion) -> bool:
-        """Check to see if a response is relevant to the chatbot's knowledge or not.
+    def check_answer_relevance(self, answer: str, unknown_prompt: str = None) -> bool:
+        """Check to see if a generated answer is relevant to the chatbot's knowledge or not.
 
         We assume we've prompt-engineered our bot to say a response is unrelated to the context if it isn't relevant.
         Here, we compare the embedding of the response to the embedding of the prompt-engineered "I don't know" embedding.
 
-        set the unk_threshold to 0 to essentially turn off this feature.
+        unk_threshold can be a value between [-1,1]. Usually, 0.85 is a good value.
```

Comment:
Can it really be negative ?

Reply:
i mean in theory cosine distance is in range [-1, 1] so i dont see why not